Formula/grafana-agent.rb

```diff
@@ -0,0 +1,80 @@
+class GrafanaAgent < Formula
+  desc "Exporter for Prometheus Metrics, Loki Logs, and Tempo Traces"
+  homepage "https://grafana.com/docs/agent/"
+  url "https://github.com/grafana/agent/archive/refs/tags/v0.21.2.tar.gz"
+  sha256 "a9deceaf1c09f19e5f44204e7d7c755dc24a5dd5a532c76c0524e762a695b590"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  on_linux do
+    depends_on "systemd" => :build
+  end
+
+  def install
+    ldflags = %W[
+      -X github.com/grafana/agent/pkg/build.Branch=HEAD
+      -X github.com/grafana/agent/pkg/build.Version=v#{version}
+      -X github.com/grafana/agent/pkg/build.BuildUser=#{tap.user}
+      -X github.com/grafana/agent/pkg/build.BuildDate=#{time.rfc3339}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags.join(" ")), "./cmd/agent"
+    system "go", "build", *std_go_args(ldflags: ldflags.join(" ")), "-o", bin/"grafana-agentctl", "./cmd/agentctl"
+
+    (bin/"grafana_agent_brew_services").write <<~EOS
+      #!/bin/bash
+      exec #{bin}/grafana-agent $(<#{etc}/grafana-agent.args)
+    EOS
+
+    (buildpath/"grafana-agent.args").write <<~EOS
+      -config.file #{etc}/grafana-agent.yaml
+      -prometheus.wal-directory #{var}/grafana-agent
```

Comment:
WAL files should not be stored directly in the grafana-agent dir.
```suggestion
      -prometheus.wal-directory #{var}/grafana-agent/prometheus/wal
```